### PR TITLE
JBIDE-24963-8 - add new testclasses for CDI 2.0

### PR DESCRIPTION
**Pull request by zcervink**
JBIDE-24963-8 - add new testclasses: 
- EjbProjectWithCDITestCDI20.class
 - ExcludeBeanTestCDI20.class
 - IBindingValidationQuickFixTestCDI20.class
 - InterceptorValidationQuickFixTestCDI20.class
 - NamedRefactoringTestCDI20.class
 - NullValuesInjectionTestCDI20.class
 - QualifierValidationQuickFixTestCDI20.class

Signed-off-by: Zbynek Cervinka <zcervink@redhat.com>

**Jenkins:** https://dev-platform-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/cdi20.itests/84/
**Jira:** https://issues.jboss.org/browse/JBIDE-24963

please review @odockal @jkopriva

### CODE REVIEW

**Comment by zcervink** [comment]
testPR

**Comment by zcervink** [comment]
testPR

**Comment by zcervink** [comment]
testPR

File: tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/beans/interceptor/cdi20/InterceptorValidationQuickFixTestCDI20.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+		}
+	}
+	
+	public InterceptorValidationQuickFixTestCDI20() {
+		CDIVersion = "2.0";
+	}
+	
+	@Before
+	public void changeDiscoveryMode(){
+		validationProvider = new InterceptorValidationProviderCDI11();
```
Comment (odockal) [inline tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/beans/interceptor/cdi20/InterceptorValidationQuickFixTestCDI20.java]: This validation provider still uses JSR-346 string in problem's message.

File: tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/beans/qualifier/cdi20/QualifierValidationQuickFixTestCDI20.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+		}
+	}
+	
+	public QualifierValidationQuickFixTestCDI20() {
+		CDIVersion = "2.0";
+	}
+	
+	@Before
+	public void changeDiscoveryMode(){
+		validationProvider = new QualifierValidationProviderCDI11();
```
Comment (odockal) [inline tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/beans/qualifier/cdi20/QualifierValidationQuickFixTestCDI20.java]: Wrong JSR number in validation provider.

**Comment by odockal** [comment]
@zcervink It is interesting how is it possible that jenkins build passes but validation provider is supposed to use wrong JSR definition. What is the actual problem's message assert?

File: tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/beans/interceptor/cdi20/InterceptorValidationQuickFixTestCDI20.java (hunk 1 above)
Comment (zcervink) [reply tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/beans/interceptor/cdi20/InterceptorValidationQuickFixTestCDI20.java]: Fixed.

File: tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/beans/qualifier/cdi20/QualifierValidationQuickFixTestCDI20.java (hunk 2 above)
Comment (zcervink) [reply tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/beans/qualifier/cdi20/QualifierValidationQuickFixTestCDI20.java]: Fixed.

**Comment by zcervink** [comment]
@odockal Indeed very interesting. I going to investigate.

**Comment by zcervink** [comment]
testPR

**Comment by odockal** [comment]
@zcervink Testss has passed on jenkins?

**Comment by zcervink** [comment]
> @zcervink Testss has passed on jenkins?

No, now I have implemented fix for the NPE, after PR JBIDE-24963-10 finishes the run I will run tests on Jenkins for this PR.

**Comment by zcervink** [comment]
testPR

**Comment by zcervink** [comment]
@odockal after some fixes the Jenkins jobs are now green: https://dev-platform-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/cdi20.itests/162/

**Comment by zcervink** [comment]
Update on this PR: All code smells removed, Jenkins - all green: https://dev-platform-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/cdi20.itests/170/

**Pull request merged**